lib/rules/object-curly-newline.js

```diff
@@ -180,7 +180,7 @@ module.exports = {
                         fix(fixer) {
                             return fixer.removeRange([
                                 openBrace.range[1],
-                                first.range[0]
+                                firstTokenOrComment.range[0]
```

Comment:
Good catch, it looks like that will get fixed to

```js
var foo = {// comment};
```

...which is a syntax error. To solve that, my recommendation would be to avoid fixing the closing-brace case if the token is a line comment. (Alternatively, since I suspect comments are sort of an edge case, another option would be to just not fix objects with comments at all.)